ledger/ledgercore/persistedacctdata_test.go

```diff
@@ -0,0 +1,698 @@
+		}
+		a.Equal(int(g.Count), len(newAssets))
+		a.Contains(newAssets, newAsset)
+		delete(newAssets, newAsset)
+		a.Equal(oldAssets, newAssets)
+	}
+
+	// prepend
+	aidx = spec[0].start - 10
+	e.Groups[0].insert(aidx, basics.AssetHolding{Amount: uint64(aidx)})
+	a.Equal(oldCount+1, e.Groups[0].Count)
+	a.Equal(aidx, e.Groups[0].MinAssetIndex)
+	a.Equal(oldDeltaMaxAssetIndex, e.Groups[0].DeltaMaxAssetIndex)
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.AssetOffsets))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Amounts))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Frozens))
+	a.Equal(basics.AssetIndex(0), e.Groups[0].groupData.AssetOffsets[0])
+	a.Equal(spec[0].start-aidx, e.Groups[0].groupData.AssetOffsets[1])
+	a.Equal(oldAssetOffsets[1:], e.Groups[0].groupData.AssetOffsets[2:])
+	checkAssetMap(aidx, e.Groups[0])
+
+	// append
+	e = genExtendedHolding(t, spec)
+	aidx = spec[0].end + 10
+	e.Groups[0].insert(aidx, basics.AssetHolding{Amount: uint64(aidx)})
+	a.Equal(oldCount+1, e.Groups[0].Count)
+	a.Equal(spec[0].start, e.Groups[0].MinAssetIndex)
+	a.Equal(uint64(aidx-spec[0].start), e.Groups[0].DeltaMaxAssetIndex)
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.AssetOffsets))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Amounts))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Frozens))
+	a.Equal(basics.AssetIndex(0), e.Groups[0].groupData.AssetOffsets[0])
+	a.Equal(oldAssetOffsets, e.Groups[0].groupData.AssetOffsets[:e.Groups[0].Count-1])
+	a.Equal(aidx-spec[0].end, e.Groups[0].groupData.AssetOffsets[e.Groups[0].Count-1])
+	checkAssetMap(aidx, e.Groups[0])
+
+	// insert in the middle
+	e = genExtendedHolding(t, spec)
+	aidx = spec[0].end - 1
+	delta := spec[0].end - aidx
+	e.Groups[0].insert(aidx, basics.AssetHolding{Amount: uint64(aidx)})
+	a.Equal(oldCount+1, e.Groups[0].Count)
+	a.Equal(spec[0].start, e.Groups[0].MinAssetIndex)
+	a.Equal(uint64(spec[0].end-spec[0].start), e.Groups[0].DeltaMaxAssetIndex)
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.AssetOffsets))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Amounts))
+	a.Equal(int(e.Groups[0].Count), len(e.Groups[0].groupData.Frozens))
+	a.Equal(basics.AssetIndex(0), e.Groups[0].groupData.AssetOffsets[0])
+	a.Equal(oldAssetOffsets[:len(oldAssetOffsets)-1], e.Groups[0].groupData.AssetOffsets[:e.Groups[0].Count-2])
+	a.Equal(oldAssetOffsets[len(oldAssetOffsets)-1]-delta, e.Groups[0].groupData.AssetOffsets[e.Groups[0].Count-2])
+	a.Equal(delta, e.Groups[0].groupData.AssetOffsets[e.Groups[0].Count-1])
+	checkAssetMap(aidx, e.Groups[0])
+}
+
+func checkHoldings(t *testing.T, g AssetsHoldingGroup) {
+	a := require.New(t)
+	aidx := g.MinAssetIndex
+	for i := 0; i < int(g.Count); i++ {
+		aidx += g.groupData.AssetOffsets[i]
+		a.Equal(uint64(aidx), g.groupData.Amounts[i])
+	}
+}
+
+// test for AssetsHoldingGroup.splitInsert
+func TestAssetHoldingSplitInsertGroup(t *testing.T) {
```

Comment:
could you add a test to cover both odd and even group sizes ?

Reply:
done